extension/src/ctp.js

```diff
@@ -61,11 +61,12 @@ function createCtpClient({
   })
 }
 
-function setUpClient() {
-  const ctpClient = createCtpClient(config.ctp)
+function setUpClient(ctpProjectKey) {
+  const ctpConfig = config.getCtpConfig(ctpProjectKey)
```

Comment:
We need to consider a cache for managing multiple client objects.

Reply:
I've checked your question in the slack channel and no answer yet. I could add a Map here for that purpose, but I'm not sure if it's useful.

Reply:
That below should be working, with this we don't need to create multiple auth flows for the same project. 
https://github.com/commercetools/nodejs/blob/aa630f042929864d0a06a6461e876ec305d5a775/types/sdk.js#L171-L180

````js
const tokenCache = {
  store: {},
  get(tokenCacheOptions) {
    return this.store[tokenCacheOptions.projectKey]
  },
  set(cache, tokenCacheOptions) {
    this.store[tokenCacheOptions.projectKey] = cache
  }
}
````

then we need to pass the tokenCache to middleware:

````js
const authMiddleware = createAuthMiddlewareForClientCredentialsFlow({
    host: config.ctp.authUrl,
    projectKey,
    credentials: {
      clientId,
      clientSecret,
    },
    fetch,
    tokenCache
  })
````

Reply:
Thanks for the link. I'm checking it but could not find this option in the docs, maybe it's a hidden feature 😄  https://commercetools.github.io/nodejs/sdk/api/sdkMiddlewareAuth.html. Anyway, a deeper look in the source code shows that if you don't pass anything, there should be a cache by default, see here: https://github.com/commercetools/nodejs/blob/aa630f042929864d0a06a6461e876ec305d5a775/packages/sdk-middleware-auth/src/client-credentials-flow.js#L19. So I guess it's already working like you suggested.

Reply:
Don't guess, default impl is here: https://github.com/commercetools/nodejs/blob/a72fd8424250fc50c84197b505a44756ceb407b0/packages/sdk-middleware-auth/src/utils.js#L1-L10

Reply:
You're right. I had a deeper and still could not understand how it really works if we have multiple clients to cache. I had a deeper look and cannot understand how it works and how we should set it up, so I just added my own cache: https://github.com/commercetools/commercetools-adyen-integration/pull/526/commits/80f8227dda50527fa060fc4aec9aacce1f817b2d

Reply:
Why you are not trying this one ? https://github.com/commercetools/commercetools-adyen-integration/pull/526#discussion_r563950126

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/526/commits/c941656fca7b9e3558a8910b57c9c2493c603939